### ISSUE TITLE
Allow for retries in the HTTP API (no streaming).

### ISSUE THREAD

**Issue by smn** [praekeltfoundation/vumi-go#959]
- [ ] The amount of retries should be configurable with a max value.
- [ ] The range of HTTP codes which should trigger a retry should be configurable (while providing a sane default like `503`)

My thinking is that we can do this with a callLater and storing a 'retry' value on the actual message being processed.

This is going to cause issues with our current way of processing 1 message at a time because we should only AMQP "ack" the message after `n` retries but don't want to have the process wait around and do nothing until `n` retries have completed.

**Comment by hodgestar** [comment]
Maybe we should dust off the re-queueing stuff and do this properly?

**Comment by smn** [comment]
Yes, thinking it through a bit more using `callLater` is a terrible idea. If we have an influx of 100 messages a second we'll build up a massive in memory backlog per worker which we cannot persist to disk during a restart nor have any control over. This would increase the likely hood of DDoSing an already struggling service because during retries we'll likely be generating 200 msg/s, 100 from the backlog & 100 from the stuff that's arriving.

We'll need to queue retries elsewhere like we did with the FailureWorker approach we tried earlier and that'll need a lot of dusting.
